urla958522fc1_spirosikmd_inmdsy_08_tgz-pJ8-TesterAppJ8/src/nl/rug/peerbox/logic/filesystem/FileRequestTask.java

```diff
@@ -0,0 +1,114 @@
+package nl.rug.peerbox.logic.filesystem;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketTimeoutException;
+import java.security.MessageDigest;
+import java.util.concurrent.TimeoutException;
+
+import nl.rug.peerbox.logic.Context;
+import nl.rug.peerbox.logic.Peer;
+
+import org.apache.commons.lang3.SystemUtils;
+import org.apache.log4j.Logger;
+
+public final class FileRequestTask implements Runnable {
+
+	private static final int SOCKET_TIMEOUT = 5000;
+
+	private static final Logger logger = Logger
+			.getLogger(FileRequestTask.class);
+
+	private final Peer h;
+	private final String filename;
+	private final Context ctx;
+	private final PeerboxFile file;
+
+	public FileRequestTask(PeerboxFile file, Context ctx) {
+		this.file = file;
+		this.h = file.getOwner();
+		this.filename = file.getFilename();
+		this.ctx = ctx;
+	}
+
+	File hide(File src) {
+		if (SystemUtils.IS_OS_WINDOWS) {
+			Process p;
+			try {
+				p = Runtime.getRuntime().exec("attrib +h " + src.getPath());
+				p.waitFor();
+			} catch (IOException | InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)